docs/src/pages/components/data-grid/pagination/pagination.md

```diff
@@ -4,65 +4,75 @@ title: Data Grid - Pagination
+### Automatic page size
 
-## Auto size
+Use the `autoPageSize` prop to auto-scale the `pageSize` to match the container height and the max number of rows that can be displayed without a vertical scroll bar.
 
-The `autoPageSize` prop allows to auto-scale the `pageSize` to match the container height and the max number of rows that can be displayed without a vertical scroll bar.
-By default, this feature is off.
+> ⚠️ You can't use both the `autoPageSize` and `autoHeight` props at the same time because `autoHeight` scales the height of the grid according to the `pageSize`.
 
-{{"demo": "pages/components/data-grid/pagination/AutoPaginationGrid.js", "bg": "inline"}}
+{{"demo": "pages/components/data-grid/pagination/PageSizeAuto.js", "bg": "inline"}}
 
-## Server-side pagination
+### Page size options
 
-By default, pagination works on the client-side.
-To switch it to server-side, set `paginationMode="server"`.
-You also need to set the `rowCount` prop so that the grid knows the total number of pages.
-Finally, you need to handle the `onPageChange` callback to load the rows for the corresponding page.
+You can configure the page size the user can choose from with the `rowsPerPageOptions` prop.
 
-{{"demo": "pages/components/data-grid/pagination/ServerPaginationGrid.js", "bg": "inline"}}
+{{"demo": "pages/components/data-grid/pagination/PageSizeCustomOptions.js", "bg": "inline"}}
+
+## Controlled page
+
+Use the `page` prop to control the size of the pages.
+
+You can use the `onPageChange` to listen to changes to the page size and update the prop accordingly.
+
+{{"demo": "pages/components/data-grid/pagination/PageControlled.js", "bg": "inline"}}
+
+## Server-side pagination
+
+By default, the pagination is handled on the client.
+This means you have to give the rows of all pages to the grid.
+If your dataset is too big, and you only want to fetch the current page, you can use server-side pagination.
 
 **Note**: For more information regarding server-side pagination in combination with controlled selection check [here](/components/data-grid/selection/#usage-with-server-side-pagination)
 
-## Cursor-based pagination
+### Basic implementation
 
-You can adapt the pagination for your cursor-based pagination.
-To do so, you just have to keep track of the next cursor associated with each page you fetched.
+- Set the prop `paginationMode` to `server`
+- Provide a `rowCount` prop to let the grid know how many pages there is
+- Add a `onPageChange` callback ot load the rows when the page changes
 
-{{"demo": "pages/components/data-grid/pagination/CursorPaginationGrid.js", "bg": "inline"}}
+{{"demo": "pages/components/data-grid/pagination/ServerPaginationGrid.js", "bg": "inline"}}
 
-## Customization
+### Cursor implementation
 
-You can customize the rendering of the pagination in the footer following [the component section](/components/data-grid/components/#pagination) of the documentation.
+You can also create a cursor-based pagination.
```

Comment:
"create" makes me feel it is a new feature, I did not get it was the server specificity

```suggestion
You can also handle servers with cursor-based pagination.
```

Reply:
Done